doc/contribute/devtheory.rst

```diff
@@ -0,0 +1,185 @@
+Developer's Theory Guide
+========================
+
+This guide is meant to give developers a comprehensive understanding of how
+DAGMC works as it exists in MOAB and the key components one may use during
+development.
+
+Contructor
+~~~~~~~~~~~~
+
+To initialize a DAGMC instance, the constructor must be called. If the
+constructor is called with no arguments, then a new instance is created. Else,
+a pointer to an existing instance can be passed in the constructor and DAGMC
+will be attached to that existing instance.
+
+DAGMC Setup
+~~~~~~~~~~~~
+
+There are three main steps to setting up a DAGMC geometry. First the storage
+space for the implicit complement is created. Then the OBB trees are built.
+And finally the sets of indices for volumes and surfaces are generated.
+See sections below for more information about implicit complements, OBBs,
+and indices.
+
+Topology
```

Comment:
This is probably the most difficult concept to convey and might merit some kind of diagram or tree.
The difference between these geometry entities represented as MOAB entity sets with parent-child relationships to these sets which in turn have a containment relationship with mesh entities that replace the analytic geometric representation. With that in mind I'd be really specific when using the word entity, including some clarifying term in front.